tensornetwork/backends/pytorch/decompositions.py

```diff
@@ -146,38 +147,43 @@ def qr_decomposition(
   Args:
     tf: The tensorflow module.
     tensor: A tensor to be decomposed.
-    split_axis: Where to split the tensor's axes before flattening into a
+    pivot_axis: Where to split the tensor's axes before flattening into a
       matrix.
 
   Returns:
     Q: Left tensor factor.
     R: Right tensor factor.
   """
 
-  left_dims = list(tensor.shape)[:split_axis]
-  right_dims = list(tensor.shape)[split_axis:]
+  left_dims = list(tensor.shape)[:pivot_axis]
+  right_dims = list(tensor.shape)[pivot_axis:]
 
   tensor = torch.reshape(tensor, (np.prod(left_dims), np.prod(right_dims)))
   q, r = torch.qr(tensor)
+  if non_negative_diagonal:
+    phases = torch.sign(torch.diagonal(r))
+    q = q * phases
+    r = torch.diag_embed(phases) @ r
```

Comment:
I think you can use broadcasting here as well:
`phases[:,None] * r` should be the same (check again though)

Reply:
Works